osu.Game.Tests/Visual/Gameplay/TestSceneOverlayActivation.cs

```diff
@@ -23,40 +24,49 @@ public override void SetUpSteps()
         [Test]
         public void TestGameplayOverlayActivation()
         {
+            AddAssert("local user playing", () => Player.LocalUserPlaying.Value);
             AddAssert("activation mode is disabled", () => Player.OverlayActivationMode == OverlayActivation.Disabled);
         }
 
         [Test]
         public void TestGameplayOverlayActivationPaused()
         {
+            AddAssert("local user playing", () => Player.LocalUserPlaying.Value);
             AddAssert("activation mode is disabled", () => Player.OverlayActivationMode == OverlayActivation.Disabled);
             AddStep("pause gameplay", () => Player.Pause());
+            AddAssert("local user not playing", () => !Player.LocalUserPlaying.Value);
             AddUntilStep("activation mode is user triggered", () => Player.OverlayActivationMode == OverlayActivation.UserTriggered);
         }
```

Comment:
This test doesn't cover resume, but adding resume here is not instant because of the existence of the resume "click orange cursor" overlay. Infrastructure to test is already in other tests (`TestScenePause`, for one) - may be wiser to move this there as in its current state it feels like it doesn't really belong in this scene basing on its name anyway.

Reply:
I placed them in this test scene given that the logic that assigns both `LocalUserPlaying` and `OverlayActivationMode` during play are in the same place in `Player`.  If one breaks, it's likely the other will also break.

If you think it would be better to put the `LocalUserPlaying` asserts somewhere that tests pause/resume functionality, the `OverlayActivationMode` asserts should probably go there too.

Alternatively I could rename `TestSceneOverlayActivation` to something like `TestScenePlayingState` that covers both, and additionally add a `LocalUserPlaying` assert to `TestScenePause`.

Reply:
i think it'll do for now.